### DISCUSSION TITLE
How to disassemble original firmware?

### DISCUSSION THREAD

**Discussion by bisak** [danielnilsson9/bbs-fw#19] [Q&A]
Hey man, I wanted to compare the disassemblies 2 different stock firmwares, I was wondering how one could disassemble an original .bin?

As a JS developer I have zero clue what's going on here, just my simple thoughts were:

Get original 52v 28A firmware
Get original 52v 30A firmware
Figure out the differences in their disassemblies
Change to 32A?
Compile and run on BBSHD?

Is this too far out / even possible?

**Comment by danielnilsson9** [comment]
Absolutely, should definitely be possible to do.
Might be enough to change the max current sent to the NEC controller.
See here https://github.com/danielnilsson9/bbshd-fw/wiki/NEC-Motor-Control-Protocol

Ghidra has a 8051 disassembler, that is what I used.

**Reply by bisak** [reply]
Interesting. That sounds good for a start. Hopefully I find some time/will to start on this.
Thank you!

**Reply by danielnilsson9** [reply]
According to my notes I had Function 0x93f2 performed this initialization in the firmware i disassembled (CRR10E4830E105001.1_190413-1.bin)

**Reply by bisak** [reply]
Amazing, thanks for the really good info. I'll definitely report back once/if I make any progress.

**Reply by bisak** [reply]
Well after spending ~2 hours on this, naive me came to the realization that this is way, WAY outside my specialty and I don't have nor the time neither the desire right now to learn all this. I managed to figure out the very basics of Ghidra and arrive at 0x93f2 but nothing really clicks. This will stay as an open project in my head and until maybe someday its time comes, I'll continue opening and soldering a 006 SMD resistor on top of the existing shunts for ~40 amps. 

Once again thanks for taking the time man, really appreciate it. I learned something new today :)

Here's it: https://i.imgur.com/kS4IVyg.png
